src/platforms/gbm-kms/server/display_helpers.cpp

```diff
@@ -144,9 +145,18 @@ mgmh::DRMHelper::open_all_devices(
 
                 [[fallthrough]];
             case EINVAL:
+                // Trying fallback
                 mir::log_warning(
-                    "Failed to detect whether device %s supports KMS, but continuing anyway",
+                    "Failed to detect whether device %s supports KMS, checking using fallback method",
                     device.devnode());
+
+                drmDev = drmModeGetResources(tmp_fd);
```

Comment:
This should probably be a `std::unique_ptr<drmModeRes>`, so we don't need to worry about remembering to clean it up.